plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenShiftProjectCache.java

```diff
@@ -88,27 +86,26 @@ public void connectionChanged(IConnection connection, String property, Object ol
 				&& (oldValue instanceof List)
 				&& (newValue instanceof List)) {
 			IOpenShiftConnection openshiftConnection = (IOpenShiftConnection) connection;
-			String key = getCacheKey(openshiftConnection);
-			Set<IProjectAdapter> adapters = getOrCreateEntry(key);
+			Set<IProjectAdapter> adapters = getCacheFor(openshiftConnection);
```

Comment:
i see a possible race-condition where 2 threads concurrently query the cache and create entries here since only #getCacheFor is locking the cache but the latter #createEntry isnt.